pkg_resources/__init__.py

```diff
@@ -2286,7 +2286,10 @@ def __init__(self, name, module_name, attrs=(), extras=(), dist=None):
         self.name = name
         self.module_name = module_name
         self.attrs = tuple(attrs)
-        self.extras = Requirement.parse(("x[%s]" % ','.join(extras))).extras
+        if extras:
+            self.extras = Requirement.parse(("x[%s]" % ','.join(extras))).extras
```

Comment:
It looks like `Requirement.parse` and underlying `parse_requirements` aren't so expensive. But anyway I'd add this check there.